Add link to commit in build details

adellibovi opened the pull request:
As title, this fixes #580 

Screenshot:
<img width="481" alt="screen shot 2018-04-07 at 12 03 23" src="https://user-images.githubusercontent.com/67916/38453800-b9b4aab2-3a5b-11e8-8a51-44704ebbaa2f.png">

adellibovi commented on app/shared/models/build.rb:
```diff
@@ -73,5 +73,10 @@ def status=(new_value)
+
+    def link_to_remote_commit
+      # TODO: The line below doesn't work for `git://` URLs (which we currently don't support anyway)
+      "#{project.repo_config.git_url}/commit/#{sha}"
```
Right now this is very "hard coded", took from the summary view.
Eventually, I would say that we should use the provider+repo_config in order to build this url, for example: GitHub uses "/commit/xx", BitBucket uses "/commits/xx", other provider would probably have a different URL too. Also, at the moment, it looks like that repo_config is strictly linked to `GitRepoConfig` (which despite the name, it seems to be for GitHub provider only, as specified by `provider_credential_type_needed`. Anyway, I only started to look into the project very recently so my understanding could be wrong :) Thoughts?

minuscorp replied:
I'd try to make a switch case depending on the current provider credential type, raise an exception when other provider credential is not github for now, but that way we avoid getting that hard-coded

taquitos replied:
@adellibovi yeah, good find: `GitRepoConfig` is tied to `GitHub`, it probably should be renamed

adellibovi replied:
@taquitos @minuscorp, What if `RepoConfig` will implement a method to return the right "link"? Of course, assuming `GitRepoConfig` is for `GitHub`. The idea is to avoid having to implement different cases of switches and instead implementing a method to the RepoConfig which is already "provider"-related. Wouldn't it be better? Also, since we have the "fullName" we may actually create the link without relying on the `git_url` and therefor fixes the `TODO` too.

taquitos replied:
Yeah, that makes sense. When we tie a Class to a provider, we can then just create a method that returns the correct link without even using a switch statement. Good idea 👍

minuscorp commented on app/features/build/views/build.erb:
```diff
@@ -25,7 +25,11 @@
-      <td>Sha</td><td><%= build.sha %></td>
+      <td>Sha</td><td>
+      <a href="<%= build.link_to_remote_commit %>" target="_blank" class="commit-sha">
+        <%= build.sha %>
```
I'm basing in my experience with other CIs that, even in the build details, the whole sha is meaningful, so i think we should use the short form whenever is possible (even in GitHub is hard to see the full commit sha 😅 )

adellibovi replied:
Jenkins shows the full hash in the build details page, I would like to hear more opinions on this (or anyway it could be done in a separately PR), personally I don't have a strong opinion on what to show.

adellibovi commented:
So, I renamed GitRepoConfig to GitHubRepoConfig and added a link_to_remote_commit there, so we avoid implementing the different cases. (I forced push on my branch so we lost those comment, sorry 🐕 )
